#Install conda and set up evaluation scripts

In [1]:
%env PYTHONPATH=


env: PYTHONPATH=


In [2]:
! wget https://repo.anaconda.com/miniconda/Miniconda3-py310_23.10.0-1-Linux-x86_64.sh
! chmod +x Miniconda3-py310_23.10.0-1-Linux-x86_64.sh
! bash ./Miniconda3-py310_23.10.0-1-Linux-x86_64.sh -b -f -p /usr/local/

--2023-12-06 06:17:17--  https://repo.anaconda.com/miniconda/Miniconda3-py310_23.10.0-1-Linux-x86_64.sh
Resolving repo.anaconda.com (repo.anaconda.com)... 104.16.131.3, 104.16.130.3, 2606:4700::6810:8303, ...
Connecting to repo.anaconda.com (repo.anaconda.com)|104.16.131.3|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 113756773 (108M) [application/x-sh]
Saving to: ‘Miniconda3-py310_23.10.0-1-Linux-x86_64.sh’

Miniconda3-py310_23 100%[===================>] 108.49M   148MB/s    in 0.7s    

2023-12-06 06:17:18 (148 MB/s) - ‘Miniconda3-py310_23.10.0-1-Linux-x86_64.sh’ saved [113756773/113756773]

PREFIX=/usr/local
Unpacking payload ...
                                                                                 
Installing base environment...





Preparing transaction: - \ | / - done
Executing transaction: | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ done
installation finished.


In [3]:
!git clone https://github.com/XavierXiao/Dreambooth-Stable-Diffusion.git
%cd Dreambooth-Stable-Diffusion

Cloning into 'Dreambooth-Stable-Diffusion'...
remote: Enumerating objects: 295, done.
remote: Total 295 (delta 0), reused 0 (delta 0), pack-reused 295
Receiving objects: 100% (295/295), 5.94 MiB | 18.20 MiB/s, done.
Resolving deltas: 100% (119/119), done.
/content/Dreambooth-Stable-Diffusion


In [4]:
%%bash
conda env create -f environment.yaml
source activate ldm

Channels:
 - pytorch
 - defaults
Platform: linux-64
Solving environment: ...working... done

Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... By downloading and using the CUDA Toolkit conda packages, you accept the terms and conditions of the CUDA End User License Agreement (EULA): https://docs.nvidia.com/cuda/eula/index.html

done
Installing pip dependencies: ...working... Ran pip subprocess with arguments:
['/usr/local/envs/ldm/bin/python', '-m', 'pip', 'install', '-U', '-r', '/content/Dreambooth-Stable-Diffusion/condaenv.92o3ijsc.requirements.txt', '--exists-action=b']
Pip subprocess output:
Obtaining taming-transformers from git+https://github.com/CompVis/taming-transformers.git@master#egg=taming-transformers (from -r /content/Dreambooth-Stable-Diffusion/condaenv.92o3ijsc.requirements.txt (line 17))
  Cloning https://github.com/CompVis/taming-transformers.git (to revision master) to ./src/taming-transformers
Ob

In [12]:
!mkdir -p models/ldm/text2img-large/
!wget -O models/ldm/text2img-large/model.ckpt https://ommer-lab.com/files/latent-diffusion/nitro/txt2img-f8-large/model.ckpt

--2023-12-06 06:29:10--  https://ommer-lab.com/files/latent-diffusion/nitro/txt2img-f8-large/model.ckpt
Resolving ommer-lab.com (ommer-lab.com)... 141.84.41.65
Connecting to ommer-lab.com (ommer-lab.com)|141.84.41.65|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 6152314307 (5.7G)
Saving to: ‘models/ldm/text2img-large/model.ckpt’

models/ldm/text2img 100%[===================>]   5.73G  30.4MB/s    in 3m 18s  

2023-12-06 06:32:28 (29.7 MB/s) - ‘models/ldm/text2img-large/model.ckpt’ saved [6152314307/6152314307]



In [14]:
%%bash
source activate ldm
python evaluation/clip_eval.py

CLIP-I :0.9833984375, CLIP-T:0.223388671875


In [15]:
!unzip /content/DDIM_50_steps.zip

Archive:  /content/DDIM_50_steps.zip
   creating: DDIM_50_steps/
  inflating: DDIM_50_steps/5.png     
  inflating: DDIM_50_steps/2.png     
  inflating: DDIM_50_steps/0.png     
  inflating: DDIM_50_steps/4.png     
  inflating: DDIM_50_steps/1.png     
  inflating: DDIM_50_steps/3.png     


In [ ]:
!unzip /content/DDIM_80_steps.zip
!unzip /content/PNDM_50_steps.zip
!unzip /content/PNDM_80_steps.zip
!unzip /content/DDIM_50_steps_prior.zip
!unzip /content/DDIM_80_steps_prior.zip
!unzip /content/PNDM_50_steps_prior.zip
!unzip /content/PNDM_80_steps_prior.zip


In [33]:
!wget https://raw.githubusercontent.com/rajlm10/dreambooth_lora_merging/main/eval.py
!mv /content/Dreambooth-Stable-Diffusion/eval.py /content/Dreambooth-Stable-Diffusion/evaluation/


--2023-12-06 07:07:27--  https://raw.githubusercontent.com/rajlm10/dreambooth_lora_merging/main/eval.py
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.109.133, 185.199.110.133, 185.199.108.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.109.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 6208 (6.1K) [text/plain]
Saving to: ‘eval.py’

eval.py             100%[===================>]   6.06K  --.-KB/s    in 0s      

2023-12-06 07:07:27 (67.7 MB/s) - ‘eval.py’ saved [6208/6208]



#DDIM No Prior Preservation

##50 steps

In [23]:
%%bash
source activate ldm

python evaluation/eval.py \
--gen_samples="/content/Dreambooth-Stable-Diffusion/DDIM_50_steps" \
--src_images="/content/raj" \
--target_text="A photo of a rraj person"

Argument values:
/content/Dreambooth-Stable-Diffusion/DDIM_50_steps
/content/raj
A photo of a rraj person

CLIP-I :0.9580078125, CLIP-T:0.206298828125


##80 Steps

In [25]:
%%bash
source activate ldm

python evaluation/eval.py \
--gen_samples="/content/Dreambooth-Stable-Diffusion/DDIM_80_steps" \
--src_images="/content/raj" \
--target_text="A photo of a rraj person"

Argument values:
/content/Dreambooth-Stable-Diffusion/DDIM_80_steps
/content/raj
A photo of a rraj person

CLIP-I :0.96630859375, CLIP-T:0.206298828125


#PNDM No Prior Preservation

##50 Steps

In [26]:
%%bash
source activate ldm

python evaluation/eval.py \
--gen_samples="/content/Dreambooth-Stable-Diffusion/PNDM_50_steps" \
--src_images="/content/raj" \
--target_text="A photo of a rraj person"

Argument values:
/content/Dreambooth-Stable-Diffusion/PNDM_50_steps
/content/raj
A photo of a rraj person

CLIP-I :0.97607421875, CLIP-T:0.206298828125


##80 steps

In [27]:
%%bash
source activate ldm

python evaluation/eval.py \
--gen_samples="/content/Dreambooth-Stable-Diffusion/PNDM_80_steps" \
--src_images="/content/raj" \
--target_text="A photo of a rraj person"

Argument values:
/content/Dreambooth-Stable-Diffusion/PNDM_80_steps
/content/raj
A photo of a rraj person

CLIP-I :0.97509765625, CLIP-T:0.206298828125


#DDIM Prior Preservation

##50 steps

In [28]:
%%bash
source activate ldm

python evaluation/eval.py \
--gen_samples="/content/Dreambooth-Stable-Diffusion/DDIM_50_steps_prior" \
--src_images="/content/raj" \
--target_text="A photo of a rraj person"

Argument values:
/content/Dreambooth-Stable-Diffusion/DDIM_50_steps_prior
/content/raj
A photo of a rraj person

CLIP-I :0.97607421875, CLIP-T:0.206298828125


##80 steps

In [29]:
%%bash
source activate ldm

python evaluation/eval.py \
--gen_samples="/content/Dreambooth-Stable-Diffusion/DDIM_80_steps_prior" \
--src_images="/content/raj" \
--target_text="A photo of a rraj person"

Argument values:
/content/Dreambooth-Stable-Diffusion/DDIM_80_steps_prior
/content/raj
A photo of a rraj person

CLIP-I :0.96826171875, CLIP-T:0.206298828125


#PNDM Prior Preservation

##50 steps

In [30]:
%%bash
source activate ldm

python evaluation/eval.py \
--gen_samples="/content/Dreambooth-Stable-Diffusion/PNDM_50_steps_prior" \
--src_images="/content/raj" \
--target_text="A photo of a rraj person"

Argument values:
/content/Dreambooth-Stable-Diffusion/PNDM_50_steps_prior
/content/raj
A photo of a rraj person

CLIP-I :0.97021484375, CLIP-T:0.206298828125


##80 steps

In [31]:
%%bash
source activate ldm

python evaluation/eval.py \
--gen_samples="/content/Dreambooth-Stable-Diffusion/PNDM_80_steps_prior" \
--src_images="/content/raj" \
--target_text="A photo of a rraj person"

Argument values:
/content/Dreambooth-Stable-Diffusion/PNDM_80_steps_prior
/content/raj
A photo of a rraj person

CLIP-I :0.96875, CLIP-T:0.206298828125


#Summary

| Scheduler | Steps | Prior Preservation | CLIP-I |
|-----------|-------|--------------------|--------|
| DDIM      | 50    | No                 | 0.9580 |
| DDIM      | 50    | Yes                | 0.9760 |
| DDIM      | 80    | No                 | 0.9663 |
| DDIM      | 80    | Yes                | 0.9683 |
| PNDM      | 50    | No                 | 0.9761 |
| PNDM      | 50    | Yes                | 0.9702 |
| PNDM      | 80    | No                 | 0.9751 |
| PNDM      | 80    | Yes                | 0.9688 |